gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -855,4 +874,486 @@ public void testListGlobalAddresses() throws InterruptedException {
+    assertNotNull(snapshot.status());
+    assertEquals(100L, (long) snapshot.diskSizeGb());
+    assertEquals(diskName, snapshot.sourceDisk().disk());
+    assertNotNull(snapshot.sourceDiskId());
+    assertNotNull(snapshot.storageBytes());
+    assertNotNull(snapshot.storageBytesStatus());
+    remoteDisk.delete();
+    diskInfo =
+        DiskInfo.of(snapshotDiskId, SnapshotDiskConfiguration.of(SnapshotId.of(snapshotName)));
+    operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    // test get disk
+    remoteDisk = compute.get(snapshotDiskId);
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(snapshotDiskId.disk(), remoteDisk.diskId().disk());
+    assertEquals(DiskInfo.CreationStatus.READY, remoteDisk.creationStatus());
+    assertNotNull(remoteDisk.creationTimestamp());
+    assertNotNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof SnapshotDiskConfiguration);
+    SnapshotDiskConfiguration remoteConfiguration = remoteDisk.configuration();
+    assertEquals(DiskConfiguration.Type.SNAPSHOT, remoteConfiguration.type());
+    assertEquals(snapshotName, remoteConfiguration.sourceSnapshot().snapshot());
+    assertEquals(100L, (long) remoteConfiguration.sizeGb());
+    assertEquals("pd-standard", remoteConfiguration.diskType().diskType());
+    assertNotNull(remoteConfiguration.sourceSnapshotId());
+    assertNull(remoteDisk.lastAttachTimestamp());
+    assertNull(remoteDisk.lastDetachTimestamp());
+    // test get disk with selected fields
+    remoteDisk = compute.get(snapshotDiskId, Compute.DiskOption.fields());
+    assertNotNull(remoteDisk);
+    assertEquals(ZONE, remoteDisk.diskId().zone());
+    assertEquals(snapshotDiskId.disk(), remoteDisk.diskId().disk());
+    assertNull(remoteDisk.creationStatus());
+    assertNull(remoteDisk.creationTimestamp());
+    assertNull(remoteDisk.id());
+    assertTrue(remoteDisk.configuration() instanceof SnapshotDiskConfiguration);
+    remoteConfiguration = remoteDisk.configuration();
+    assertEquals(DiskConfiguration.Type.SNAPSHOT, remoteConfiguration.type());
+    assertEquals(snapshotName, remoteConfiguration.sourceSnapshot().snapshot());
+    assertNull(remoteConfiguration.sizeGb());
+    assertEquals("pd-standard", remoteConfiguration.diskType().diskType());
+    assertNull(remoteDisk.<SnapshotDiskConfiguration>configuration().sourceSnapshotId());
+    assertNull(remoteDisk.lastAttachTimestamp());
+    assertNull(remoteDisk.lastDetachTimestamp());
+    operation = remoteDisk.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.get(snapshotDiskId));
+    operation = snapshot.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.getSnapshot(snapshotName));
+  }
+
+  @Test
+  public void testListDisksAndSnapshots() throws InterruptedException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method testListDisksAndSnapshots() has an NPath complexity of 256](https://www.codacy.com/app/mziccard/gcloud-java/file/2383968105/issues/source?bid=3140968&fileBranchId=3223208#l1073)